Tests/UnitTests/Scripts/engine-compat-tests.ts

```diff
@@ -0,0 +1,413 @@
+// Tests specifically for V8/JSC engine compatibility and Android XR readiness
```

Comment:
I don't understand what these tests have to do with Android XR. Can you elaborate on the purpose of these tests?